great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2522,13 +2537,18 @@ def add_or_update_profiler(
             )
 
         if existing_profiler:
-            self.update_profiler(existing_profiler)
-            return self.get_profiler(
+            self.delete_profiler(
                 name=existing_profiler.name, ge_cloud_id=existing_profiler.ge_cloud_id
             )
+            if id is None:
+                id = existing_profiler.ge_cloud_id
+            if config_version is None:
+                config_version = existing_profiler.config_version
+            if rules is None:
+                rules = {rule.name: rule.to_dict() for rule in existing_profiler.rules}
+            if variables is None:
+                variables = convert_variables_to_dict(existing_profiler.variables)
```

Comment:
Special cases since the input objects go through transformations in the __init__